### PR TITLE
doc: net: Add HTTP client library page

### LINKED ISSUE #46267: docs: include/zephyr/net/http_client.h is not documented anywhere
**Describe the bug**

This file contains the following doxygen information as of 5c6aaf709c:

```c
/**
 * @brief HTTP client API
 * @defgroup http_client HTTP client API
 * @ingroup networking
 * @{
 */
```

However, this doxygen group is not mentioned anywhere in the doc/ tree, making references to APIs in side of it, such as the following .rst snippet, fail to work properly:

```
An attempt to reference :c:func:`http_client_req`
```

**To Reproduce**
1. cd doc
2. make html
3. notice doc/releases/release-notes-3.1.rst builds making the above reference do not resolve

**Expected behavior**

Since this API is mentioned in the release notes, it should be documented.

**Impact**

Poor release notes, missing networking documentation (what else are we missing?)

### PR DESCRIPTION
**Pull request by rlubos**
Add brief documentation for the HTTP client library and reference its
API.

Fixes #46267

Signed-off-by: Robert Lubos <robert.lubos@nordicsemi.no>

### CODE REVIEW

**Review by jukkar — DISMISSED** [review]

File: doc/connectivity/networking/api/http.rst [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+Overview
+********
+
+Zephyr provides a HTTP client library which allows to send HTTP requests and
+parse HTTP responses. The library communicates over socket API, it does not
+however create sockets on its own. The application is responsible for creating
+a socket and passing it to the library. Therefore, depending on the application
+needs, the library can communicate either over plain TCP socket (HTTP) or
+over TLS socket (HTTPS).
```
Comment (carlescufi) [inline doc/connectivity/networking/api/http.rst:18]:
```suggestion
Zephyr provides an HTTP client library which allows to send HTTP requests and
parse HTTP responses. The library communicates over the sockets API, it does not
however create sockets on its own. The application is responsible for creating
a socket and passing it to the library. Therefore, depending on the application's
needs, the library can communicate either over a plain TCP socket (HTTP) or
over a TLS socket (HTTPS).
```

File: doc/connectivity/networking/api/http.rst [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+     * to the HTTP server.
+     */
+    ret = http_client_req(sock, &req, 5000, NULL);
+
+If the server sent a response for the request, it will be provided to the
+application through the response callback registered in the request structure.
+The response may be provided in chunks, therefore the application shall be
+prepared for for that. The callback function, along with the structure
+containing the response data, provide information whether more data is expected
+to receive or not. A very simple response handling function is presented below:
```
Comment (carlescufi) [inline doc/connectivity/networking/api/http.rst:49]:
```suggestion
If the server responds to the request, the response will be provided to the
application through the response callback registered in the request structure.
The response may be provided in chunks, therefore the application shall be
prepared for that. The callback function, along with the structure
containing the response data, provides information about whether more data is expected
to be received. A very simple response handling function is presented below:
```

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: doc/connectivity/networking/api/http.rst [hunk 3]
```diff
@@ -0,0 +1,72 @@
+.. _http_interface:
+
+HTTP
+####
```
Comment (FrancescoSer) [inline doc/connectivity/networking/api/http.rst:4]:
```suggestion
HTTP client
###########
```

File: doc/connectivity/networking/api/http.rst [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+Overview
+********
+
+Zephyr provides an HTTP client library which allows to send HTTP requests and
+parse HTTP responses. The library communicates over the sockets API, it does not
+however create sockets on its own. The application is responsible for creating
+a socket and passing it to the library. Therefore, depending on the application's
+needs, the library can communicate either over a plain TCP socket (HTTP) or
+over a TLS socket (HTTPS).
```
Comment (FrancescoSer) [inline doc/connectivity/networking/api/http.rst:18]:
```suggestion
The HTTP client library allows you to send HTTP requests and
parse HTTP responses. The library communicates over the sockets
API but it does not create sockets on its own.

The application must be responsible for creating a socket and passing it to the library.
Therefore, depending on the application's needs, the library can communicate over
either a plain TCP socket (HTTP) or a TLS socket (HTTPS).
```

File: doc/connectivity/networking/api/http.rst [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+Sample Usage
+************
+
+The overall API of the HTTP client library is simple and consists of a single
+function. It's important to setup the request structure correctly, for example:
```
Comment (FrancescoSer) [inline doc/connectivity/networking/api/http.rst:24]:
```suggestion
The API of the HTTP client library has a single function.

The following is an example of a request structure created correctly:
```

File: doc/connectivity/networking/api/http.rst [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+     * to the HTTP server.
+     */
+    ret = http_client_req(sock, &req, 5000, NULL);
+
+If the server responds to the request, the response will be provided to the
+application through the response callback registered in the request structure.
+The response may be provided in chunks, therefore the application shall be
+prepared for that. The callback function, along with the structure containing
+the response data, provides information about whether more data is expected
+to be received. A very simple response handling function is presented below:
```
Comment (FrancescoSer) [inline doc/connectivity/networking/api/http.rst:49]:
```suggestion
If the server responds to the request, the library provides the response to the
application through the response callback registered in the request structure.
As the library can provide the response in chunks, the application must be able
to process these.

Together with the structure containing the response data, the callback function
also provides information about whether the library expects to receive more data.

The following is an example of a very simple response handling function:
```

**Review by FrancescoSer — CHANGES_REQUESTED** [review]
Just a few suggestions

**Comment by rlubos** [comment]
@FrancescoSer Thanks, I've applied your suggestions.

**Review by FrancescoSer — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**